catalog/spatial/registry/registry-rest-endpoint/src/test/java/org/codice/ddf/registry/rest/endpoint/report/RegistryReportBuilderTest.java

```diff
@@ -0,0 +1,455 @@
+        Document htmlDoc = xmlFileToDocument(html);
+        assertSummaryReportValues(htmlDoc);
+    }
+
+    @Test
+    public void testSummaryReportWithSparseRegistryPackage() throws Exception {
+        MetacardImpl metacard = convertXmltoMetacard("/csw-sparse-registry-package.xml");
+        String html = reportHelper.getSummaryHtmlFromMetacard(metacard);
+        Document htmlDoc = xmlFileToDocument(html);
+        assertSparseSummaryReportValues(htmlDoc);
+    }
+
+    @Test
+    public void testSummaryReportWithBareRegistryPackage() throws Exception {
+        MetacardImpl metacard = convertXmltoMetacard("/csw-bare-registry-package.xml");
+        String html = reportHelper.getSummaryHtmlFromMetacard(metacard);
+        Document htmlDoc = xmlFileToDocument(html);
+        String organizationTitle = htmlDoc.getElementsByTagName("h2")
+                .item(0)
+                .getTextContent();
+        assertThat(organizationTitle, is("Summary"));
+        assertThat(htmlDoc.getElementsByTagName("table")
+                .item(0)
+                .getTextContent(), is("            "));
+    }
+
+    @Test
+    public void testOrganizationReportWithSimplifiedRegistryPackage() throws Exception {
+        RegistryPackageType registryPackage = getRegistryPackageFromResource(
+                "/csw-simplified-registry-package.xml");
+        String html = reportHelper.getHtmlFromRegistryPackage(registryPackage,
+                reportHelper.ORGANIZATIONS);
+        Document htmlDoc = xmlFileToDocument(html);
+        assertOrganizationReportValues(htmlDoc);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void testOrganizationReportWithBareRegistryPackage() throws Exception {
+        RegistryPackageType registryPackage = getRegistryPackageFromResource(
+                "/csw-bare-registry-package.xml");
+        String html = reportHelper.getHtmlFromRegistryPackage(registryPackage,
+                reportHelper.ORGANIZATIONS);
+        Document htmlDoc = xmlFileToDocument(html);
+
+        String organizationTitle = htmlDoc.getElementsByTagName("h2")
+                .item(0)
+                .getTextContent();
+        assertThat(organizationTitle, is("Organizations"));
+
+        htmlDoc.getElementsByTagName("table")
+                .item(0)
+                .getTextContent();
+    }
+
+    @Test
+    public void testFullReportWithSimplifiedRegistryPackage() throws Exception {
+        RegistryPackageType registryPackage = getRegistryPackageFromResource(
+                "/csw-simplified-registry-package.xml");
+        String html = reportHelper.getHtmlFromRegistryPackage(registryPackage, reportHelper.REPORT);
+        Document htmlDoc = xmlFileToDocument(html);
+        assertFullReportValues(htmlDoc);
+    }
+
+    private void assertSummaryReportValues(Document htmlDoc) {
+        String organizationName = htmlDoc.getElementsByTagName("strong")
```

Comment:
This looks too brittle for any changes to the template.  The tests are too coupled to the exact template.  It might be useful to use an HTML parser with more predictable selectors.  For example, you could add an organization css class for the element containing organization information and another css class for the organization name.  Then you could use something like jsoup to select on `doc.select("organization name")`.  

https://jsoup.org/cookbook/extracting-data/selector-syntax